### ISSUE TITLE
Escaping arguments on Windows

### ISSUE THREAD

**Issue by jeroen** [jeroen/sys#4]
The `lpCommandLine` from `CreateProcess` is not properly escaped. It seems that `CommandLineToArgvW` which Windows programs use to parse the cmd line back into args, drops quotes:

```r
exec_wait(R.home('bin/R'), c('-e', 'stop("this is an error")'))
```

```
> stop(this is an error)
Error: unexpected symbol in "stop(this is"
```

This works (but it shouldn't).

```r
exec_wait(R.home('bin/R'), c('-e', ' "x = rnorm(10); print(x)" '))
```

But not this:

```r
exec_wait(R.home('bin/R'), c('-e', 'x = rnorm(10); print(x)'))
```

The behavior for these examples should be the same as on linux/mac.

**Comment by jeroen** [comment]
The same happens for `base::system2`. I don't really know if this is a bug just a difference between Unix and Windows how arguments get interpreted.

@gaborcsardi how do you deal with this in [callr](https://github.com/r-pkgs/callr)?

**Comment by jimhester** [comment]
`?shQuote` has a pretty good discussion on the two types of quoting for windows.

**Comment by gaborcsardi** [comment]
I seem to have missed this....

So, `callr` does not really need to deal with this, because the user does not supply arguments, at least mostly not.

But `processx` seems to have the same issue:

```r
> processx::run(R.home("bin/R"), c("-q", "-e", 'print(111)'))
$status
[1] 0

$stdout
[1] "> print(111)" "[1] 111"      "> "           "> "          

$stderr
character(0)

> processx::run(R.home("bin/R"), c("-q", "-e",'print("hello")'))
Error in processx::run(R.home("bin/R"), c("-q", "-e", "print(\"hello\")")) : 
  System command error
```

**Comment by gaborcsardi** [comment]
Btw. as a workaround, using single quotes works: 

```r
> processx::run(R.home("bin/R"), c("-q", "-e", "print('hello')"), echo = TRUE)
Running "C:/PROGRA~1/R/R-33~1.2/bin/R" "-q" "-e" "print('hello')" 
- > print('hello')
- [1] "hello"
- > 
- > 
$status
[1] 0

$stdout
[1] "> print('hello')" "[1] \"hello\""    "> "               "> "              

$stderr
character(0)
```

But it would be nice to fix this ....

**Comment by gaborcsardi** [comment]
Single quotes also work with `sys`, only double quotes are problematic...

```r
❯ exec_wait(R.home("bin/R"), c("-q", "-e",'print("hello")'))
> print("hello")
[1] "hello"
>
>
[1] 0
```

**Comment by gaborcsardi** [comment]
Single quotes also work with `sys`, only double quotes are problematic:

```r
> exec_wait(R.home("bin/R"), c("-q", "-e", "print('hello')"))
> print('hello')
[1] "hello"
```

**Comment by gaborcsardi** [comment]
Btw. this is how libuv quotes each argument by default:
https://github.com/libuv/libuv/blob/8658ef06c2aec1d7ea4cdaca3bbb709f73d3f730/src/win/process.c#L448-L518

**Comment by jeroen** [comment]
Fixed in 2.0: https://github.com/jeroen/sys/blob/master/R/quote.R
